keanu-python/keanu/algorithm/sampling.py

```diff
@@ -170,3 +188,50 @@ def _samples_generator(sample_iterator: JavaObject, vertices_unwrapped: JavaList
                 traces = []
 
         yield sample
+
+
+def __create_single_indexed_samples(network_samples: JavaObject, vertices_unwrapped: JavaList) -> Dict:
```

Comment:
If type annotations + mypy is causing grief, I think it'll be useful to write some documentation

Reply:
Added better mypy